stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -50,17 +56,24 @@ import java.io.File
 import java.util.Date
 
 public class ChatClient internal constructor(
-    private val config: ChatClientConfig,
+    public val config: ChatClientConfig,
     private val api: ChatApi,
     private val socket: ChatSocket,
-    private val notifications: ChatNotifications
-) {
+    private val notifications: ChatNotifications,
+    public var appContext: Context,
```

Comment:
Because we store an instance of the `ChatClient` into the companion object, it is similar than store it statically in Java. And here we store a context into an static reference, that could produce a memory leak. We should try to avoid to store `context` properies into our classes

Reply:
The easy fix here is to pass the context multiple times to the 3 builders. Is there another solution we should consider?